app/helpers/validation_helper.php

```diff
@@ -0,0 +1,21 @@
+<?php
+function validate_between($str, $min, $max)
+{
+    $length = mb_strlen($str); // count characters based on encoding
+    return $length <= $max && $length >= $min;
+}
+
+function validate_username($str)
+{
+    return preg_match('/^[[:alnum:]_]*$/', $str);
+}
+
+function validate_unique_name($username)
+{
+    return !User::getByUsername($username);
```

Comment:
The transaction is unnecessary po since you are only writing to one table. 
Besides the order of process is weird po, normally you validate input and if it is correct you then insert it in the database. Without the transaction, you cannot rollback if there is an invalid input.